x-pack/test/functional/services/ml/expanded_job_details.ts

```diff
@@ -0,0 +1,124 @@
+          await testSubjects.click('clearSearchButton');
+
+        await testSubjects.click(jobTable.detailsSelector(jobId, 'euiCollapsedItemActionsButton'));
+        await testSubjects.click('mlAnnotationsActionEdit');
+
+        await testSubjects.setValue('mlAnnotationsFlyoutTextInput', newAnnotation, {
+          clearWithKeyboard: true,
+        });
+        await testSubjects.click('annotationFlyoutUpdateOrCreateButton');
+      });
+
+      await jobTable.withDetailsOpen(jobId, async () => {
+        await jobTable.openAnnotationsTab(jobId);
+        if (await testSubjects.exists('clearSearchButton'))
+          await testSubjects.click('clearSearchButton');
+        const visibleText = await testSubjects.getVisibleText(
+          jobTable.detailsSelector(jobId, 'mlAnnotationsColumnAnnotation')
+        );
+        expect(visibleText).to.be(newAnnotation);
+      });
+    },
+
+    async assertDataFeedFlyout(jobId: string): Promise<void> {
+      await jobTable.withDetailsOpen(jobId, async () => {
+        await jobTable.openAnnotationsTab(jobId);
+        if (await testSubjects.exists('clearSearchButton'))
+          await testSubjects.click('clearSearchButton');
+        await testSubjects.click(jobTable.detailsSelector(jobId, 'euiCollapsedItemActionsButton'));
+        await testSubjects.click('mlAnnotationsActionViewDatafeed');
+        await testSubjects.existOrFail('mlAnnotationsViewDatafeedFlyoutChart');
+        const visibleText = await testSubjects.getVisibleText(
+          'mlAnnotationsViewDatafeedFlyoutTitle'
+        );
+        expect(visibleText).to.be(`Datafeed chart for ${jobId}`);
+        await testSubjects.click('euiFlyoutCloseButton');
+      });
+    },
+
+    async openModelSnapshotTab(jobId: string) {
+      await retry.tryForTime(3_000, async () => {
+        await jobTable.ensureDetailsOpen(jobId);
+        await testSubjects.click(jobTable.detailsSelector(jobId, 'mlJobListTab-modelSnapshots'));
+      });
+    },
+
+    async assertModelSnapshotManagement(jobId: string): Promise<void> {
+      await jobTable.withDetailsOpen(jobId, async () => {
+        await this.openModelSnapshotTab(jobId);
+        await testSubjects.existOrFail('mlADModelSnapShotRevertButton');
+        await testSubjects.existOrFail('mlADModelSnapShotsEditButton');
+      });
+    },
+
+    async selectAllJobs(): Promise<void> {
+      await testSubjects.click('checkboxSelectAll');
+    },
+
+    async assertJobListMultiSelectionText(expectedMsg: string): Promise<void> {
+      const visibleText = await testSubjects.getVisibleText('~mlADJobListMultiSelectActionsArea');
+      expect(visibleText).to.be(expectedMsg);
+    },
+
+    async assertColumnState(buttonText: string, columnJobState: string): Promise<void> {
+      await find.clickByButtonText(buttonText);
+      const visibleText = await testSubjects.getVisibleText('mlJobListColumnJobState');
+      expect(visibleText).to.be(columnJobState);
+    },
```

Comment:
1. This should rather be a method of the `job_table` service as it's not related to the expanded row.
2. The method name doesn't really reflect the action, maybe make it something like `filterByState` and then also make sure that only available quick filter buttons can be passed as parameter or so?